### PR TITLE
Refactor multiple classes to use ExtendedImage

### PR DESCRIPTION
**Pull request by martinzlocha**
Created a general ExtendedImage class which has functions like save() and later will have other functions as well. This will be very useful when we will start calculating indentation etc.

### CODE REVIEW

File: WLC/__main__.py [hunk 1]
```diff
@@ -1,10 +1,12 @@
 from WLC.code_executor import CodeExecutor
 from WLC.image_processing.camera import Camera
+from WLC.image_processing.image_preprocessor import ImagePreprocessor
```
Comment (mishrabhinav) [inline WLC/__main__.py]: wouldn't it be shorter if we just called it `WLC.image_processing.preprocessor`?

File: WLC/image_processing/image_preprocessor.py [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import cv2
+
+
+class ImagePreprocessor:
+    def __init__(self):
```
Comment (mishrabhinav) [inline WLC/image_processing/image_preprocessor.py]: Not a python expert but this doesn't seem to do anything, so remove it?

**Review by mishrabhinav — COMMENTED** [review]
@martinzlocha I suggested a few changes but they are merely cosmetic otherwise LGTM.

File: WLC/__main__.py (hunk 1 above)
Comment (martinzlocha) [reply WLC/__main__.py]: Yeah will change.

File: WLC/image_processing/image_preprocessor.py (hunk 2 above)
Comment (martinzlocha) [reply WLC/image_processing/image_preprocessor.py]: Paul added this to all other classes so will keep it just for consistency.

**Pull request merged**